Rubberduck.Core/UI/Inspections/InspectionResultsViewModel.cs

```diff
@@ -232,6 +234,95 @@ public bool GroupByLocation
             }
         }
 
+        public bool _filterInspectionsByHint;
+        public bool FilterInspectionsByHint
+        {
+            get =>_filterInspectionsByHint;
+            set
+            {
+                if (_filterInspectionsByHint == value) { return; }
+
+                FilterResults(filterByHint: value);                
+                OnPropertyChanged();
+            }
+        }
+
+        private bool _filterInspectionsBySuggestion;
+        public bool FilterInspectionsBySuggestion
+        {
+            get => _filterInspectionsBySuggestion;
+            set
+            {
+                if (_filterInspectionsBySuggestion == value) { return; }
+
+                FilterResults(filterBySuggestion: value);
+                OnPropertyChanged();
+            }
+        }
+
+        private bool _filterInspectionsByWarning;
+        public bool FilterInspectionsByWarning
+        {
+            get => _filterInspectionsByWarning;
+            set
+            {
+                if (_filterInspectionsByWarning == value) { return; }
+
+                FilterResults(filterByWarning: value);
+                OnPropertyChanged();
+            }
+        }
+
+        private bool _filterInspectionsByError;
+        public bool FilterInspectionsByError
+        {
+            get => _filterInspectionsByError;
+            set
+            {
+                if (_filterInspectionsByError == value) { return; }
+
+                FilterResults(filterByError: value);
+                OnPropertyChanged();
+            }
+        }
+
+        public void FilterResults(bool filterByHint = false, bool filterBySuggestion = false, bool filterByWarning = false, bool filterByError = false)
```

Comment:
Instead of having individual parameters, why not pass in an `IEnumerable<CodeInspectionSeverity>`?

Reply:
That's some nasty method signature right there.. 😈